r2/errors.go

```diff
@@ -60,3 +60,11 @@ type NotFoundError struct{ msg string }
 func NewNotFoundError(msg string) NotFoundError { return NotFoundError{msg: msg} }
 
 func (e NotFoundError) Error() string { return e.msg }
+
+// AuthError represents an error due to missing or invalid auth information.
```

Comment:
Should this be moved to the `auth` package instead? Doesn't seem like this is r2 specific.